src/snapshot_tests/scene_test.rs

```diff
@@ -0,0 +1,192 @@
+use anyhow::Result;
+use compositor_common::{
+    scene::{InputId, NodeId, OutputId, Resolution, SceneSpec},
+    Frame,
+};
+use compositor_render::{FrameSet, Renderer};
+use std::{collections::HashSet, fmt::Display, fs, path::PathBuf, sync::Arc, time::Duration};
+
+use super::{
+    test_case::TestInput,
+    utils::{are_snapshots_equal, frame_to_rgba, SNAPSHOTS_DIR_NAME},
+};
+
+pub struct SceneTest {
```

Comment:
This struct does not serve any purpose anymore. This abstraction does not help anything just divides implementation in 2 parts.